subprojects/composite-builds/src/integTest/groovy/org/gradle/integtests/composite/CompositeBuildCommandLineArgsIntegrationTest.groovy

```diff
@@ -143,4 +143,25 @@ includeBuild '../buildB'
         executed ":buildB:jar"
     }
 
+    def "does not execute task actions when dry run specified on composite build"() {
+        given:
+        dependency 'org.test:buildB:1.0'
+
+        when:
+        execute(buildA, ":build", ["--dry-run"])
+
+        then:
+        result.normalizedOutput.contains ":compileJava SKIPPED\n" +
```

Comment:
Instead of verifying the whole output I think it would be less brittle if we could assert the outcome of the tasks with the `skipped` method.